### DISCUSSION TITLE
Using General Resource Form(Form ) : Form, is there a static function to get the edited object

### DISCUSSION THREAD

**Discussion by ConsistentClient** [filamentphp/filament#5552] [General]
Dear Team,

Please i am newbie to filamentphp, and love it so far. 
I am rendering a form on Edit/Create using standard:

    public static function form(Form $form): Form {} 

and this part works. 
However, i need make my form change based on which WorkOrder I am editing. for example:

http://localhost:8000/admin/orders/01gqpe79gp7c2de4rhyq5ejfaj/edit

if there a static function i can use inside form(Form $form) : Form {} to give me the value i am editing which is "01gqpe79gp7c2de4rhyq5ejfaj" ? 

I tried to lookup different discussion but could not find any answers.

thank you
Excellent work,

**Comment by danharrin** [comment]
What are you trying to do based on the current record?

**Reply by ConsistentClient** [reply]
Thank you very much for responding.

The form will have different controls based on the current record. 

the record (which is an order) has multiple states: 
For example if current record is a new order, we will need a set of data to be filled first (like name, addres ...etc)
then the state of the order goes to next states, and next states, other data fields needs to be filled

**Reply by danharrin** [reply]
You can use `->hidden(fn ($record) => $record->status === '...')`

**Reply by ConsistentClient** [reply]
this is an excellent idea. i guess i will use the function to show and hide fields.
but is there anyway we can get the record id which was passed in the route  ? 
"http://localhost:8000/admin/orders/01gqpe79gp7c2de4rhyq5ejfaj/edit", can we get the id 01gqpe79gp7c2de4rhyq5ejfaj ? 

if i can get that value, the code for the form objects will be much easier to read and manage.

**Reply by danharrin** [reply]
`$record` is the model instance, `$record->id` is the ID from the route

**Reply by ConsistentClient** [reply]
Please if you can tell me the function to call. this is what i tried:

    public static function form(Form $form): Form
    {
        dd( $record->id );

which fails because $record is not defined 
I am guessing i will have to do this in the EditRecord class, and not in the Resource, correct ?

this seems to work: 
inside the EditRecord
    protected function getFormSchema(): array {
    $id = $this->record->id
 ....
}

Is this correct way to do it ? 


thank you,

**Reply by ConsistentClient** [reply]
This seems to work for me, is to use the EditRecord plass and do the getFormSchema() function
I want to say again, thank you, i love the filamentphp, and as i use it i love it even more.
well done so far,
thank you,
